### DISCUSSION TITLE
Moving part of a subtree of branches, using branch names?

### DISCUSSION THREAD

**Discussion by claytonrcarter** [arxanas/git-branchless#259] [Q&A] [answered]
Hi, I'm still new to git-branchless but so far it's been great. I'm currently using v 0.3.8 in a git repo w/ ~13k commits. I'm using it with branches (we use GitLab and merge requests, which require branches) and we often have braches stacked on top of each other.

The `git move` docs say this:

```
    -b, --base <BASE>
            A commit inside a subtree to move. The entire subtree, starting from the main branch,
            will be moved, not just the commits descending from this commit
```

Note the part about "starting from the main branch". **Is there a way to config what branch this starts from?**

For example, if I have the following branch structure (where each branch could have many commits):

```
main - a - b - c
```

And I want to move `b` and `c` onto main, but leave `a` as is. Like this:

```
main - b - c
   \- a
```

I am doing this now via `git move -s <sha of first commit in b> -d main`, but it would be really handy to be able to continue using branch names instead of looking up SHAs. I tried something like `git move --source a --base c --dest main` thinking that perhaps this might select the commits between `a` & `c`, but no luck.

Any tricks or tips that I might be missing?

And thanks again for this tool. It's wonderful!

**Comment by arxanas** [comment]
Hi @claytonrcarter, I'm glad that you're finding git-branchless useful!

In my workflows, I actually do end up copy-pasting the commit hash for various operations (which is not ideal). In this specific case, does `b` refer to a commit hash or a branch name? If it's a branch name, you should still be able to run

```
$ git move -s b -d main
```

to move the stack starting from `b`.

If not, some notes:

- There isn't currently a good way of selecting `def456` and its descendants when your stack looks like this: `abc123 branch-1 -> def456 -> ghi789 branch-2` other than via commit hash. #175 or https://github.com/arxanas/git-branchless/issues/177#issuecomment-955111132 might be able to help in this case in the future.
- There currently isn't a good way to move a commit without moving its descendants. https://github.com/arxanas/git-branchless/issues/176 would let us do this.

I haven't had a lot of time to do feature work lately, but if you're interested in implementing one of the above, I'm happy to set up a call to walk you through the codebase and figure out what needs to be done.

**Reply by claytonrcarter** [reply]
Hi, thank you for the reply, and sorry I wasn't more clear! I held off on responding sooner because it wasn't clear to me that what I wanted to do wasn't actually possible, but I think now I've confirmed that it's not, so here goes again!

In my example above, I intended letters `a`, `b` and `c` to be branches each containing multiple commits, so your comment about selecting `def456` is exactly my use case. Though it's not terribly onerous to use a commit hash, it would be very handy (again, in my use cases w/ lots of branches) to be able to do something like `git move --branch c --dest main --stop-at a` or something similar.

Thanks again and I appreciate the offer to show me the code base. If I get some time, I may try to tinker on something; either something like this or #176. I have been wanting to explore Rust; perhaps this is the excuse I need!

**Reply by arxanas** [reply]
Unfortunately, there's not something like that directly, so we'd have to build that feature.

**Comment by arxanas** [comment, marked as answer]
For the future reader, it's now possible to run something like `git move -s 'children(a)' -d 'main()'` to move all descendant commits/branches of `a`.
